### PR TITLE
Ensure native strings in environ

### PR DESCRIPTION
**Pull request by asottile**

### CODE REVIEW

**Comment by asottile** [comment]
I still need to actually check this on windows -- but I think this fixes it :)

**Comment by asottile** [comment]
can confirm this fixes things, however it's insufficient to get tox working on python2.7 on windows:

https://github.com/tox-dev/tox/issues/1315

**Comment by asottile** [comment]
@gaborbernat it looks like azure pipelines isn't actually testing python2.7 🙃

**Comment by gaborbernat** [comment]
what 😮 

```
py27 installed: apipkg==1.5,atomicwrites==1.3.0,attrs==19.1.0,colorama==0.4.1,coverage==4.5.3,execnet==1.6.0,filelock==3.0.12,flaky==3.5.3,freezegun==0.3.11,more-itertools==7.0.0,pathlib2==2.3.3,pluggy==0.11.0,psutil==5.6.2,py==1.8.0,pytest==4.5.0,pytest-cov==2.7.1,pytest-forked==1.0.2,pytest-mock==1.10.4,pytest-randomly==1.2.3,pytest-xdist==1.28.0,python-dateutil==2.8.0,six==1.12.0,toml==0.10.0,tox==3.12.1.dev1+gf0808f7,virtualenv==16.6.0,wcwidth==0.1.7
py27 run-test-pre: PYTHONHASHSEED='879'
py27 run-test: commands[0] | pytest --cov 'D:\a\1\s\.tox\py27\Lib\site-packages/tox' --cov-config 'D:\a\1\s/tox.ini' --junitxml 'D:\a\1\s\.tox/junit.py27.xml' -n=auto .
============================= test session starts =============================
platform win32 -- Python 3.7.2, pytest-4.5.0, py-1.8.0, pluggy-0.11.0 -- d:\a\1\s\.tox\py27\scripts\python.exe
cachedir: .tox\py27\.pytest_cache
```

what's this magic 🤔

File: tests/unit/session/test_session.py [hunk 1]
```diff
@@ -285,6 +285,10 @@ def assert_popen_env(res):
         if tox_id != "GLOB":
             assert env["TOX_ENV_NAME"] == tox_id
             assert env["TOX_ENV_DIR"] == os.path.join(res.cwd, ".tox", tox_id)
+        # ensure native strings for environ for windows
+        for k, v in env.items():
+            assert type(k) is str, (k, v, type(k))
```
Comment (luzfcb) [inline tests/unit/session/test_session.py]: @asottile 
A technical question: Why not use `isinstance(k, str)` instead `type`?

Comment (asottile) [reply tests/unit/session/test_session.py]: `type(...)` is a stronger assertion here than `isinstance` -- I'm guarding against subclasses of `str`

Comment (luzfcb) [reply tests/unit/session/test_session.py]: Thanks to the explanation.

```python
In [1]: class foostr(str):
   ...:     pass
   ...:
   ...:

In [2]: x = str("bar")

In [3]: y = foostr("bar")

In [4]: type(x)
Out[4]: str

In [5]: type(y)
Out[5]: __main__.foostr

In [6]: isinstance(x, str)
Out[6]: True

In [7]: isinstance(y, str)
Out[7]: True

In [8]: type(x) is str
Out[8]: True

In [9]: type(y) is str
Out[9]: False

```

**Pull request merged**